network-api/networkapi/wagtailpages/static/js/directory-listing-filters.js

```diff
@@ -0,0 +1,145 @@
+(function() {
+  // start of function
+
+  const profileCache = {};
+  const filters = document.querySelectorAll(`.profile-directory .fellowships-directory-filter .filter-option label`);
+  const profileContainer = document.querySelector(`.profiles .row`);
+  const { profileType, programType } = document.querySelector(`.profiles`).dataset;
+  const API_ENDPOINT = document.querySelector(`[data-api-endpoint]`).dataset.apiEndpoint;
+
+  /**
+   * After initial page load, the filter buttons are responsible for
+   * fetching results "per filter entry".
+   */
+  function getData(programYear) {
+    // set up a url for performing an API call:
+    let url = API_ENDPOINT;
+
+    // build and append the query arguments
+    let query = [
+      profileType ? `profile_type=${profileType}` : false,
+      programType ? `program_type=${programType}` : false,
+      programYear ? `program_year=${programYear}` : false
+    ].filter(v => v).join('&');
+
+    // and then perform our API call using the Fetch API
+    return new Promise(
+      (resolve, reject) => {
+        return fetch(`${url}&${query}`)
+               // conver the resonse from JSON to real data
+               .then(response => response.json())
+               // all went well? resolve.
+               .then(obj => resolve(obj))
```

Comment:
It's been a while: what's `resolve(obj)` do here?

Reply:
It's the "trigger `then` handling" function, with `reject` being the "trigger `catch` handling" function, which in this case it's the thing that lets https://github.com/mozilla/foundation.mozilla.org/pull/2701/files/bcbe62a6790f1f96304d02177168d8b875acb22b#diff-70b039d3f247fb0182821822739840d5R123 happen

Reply:
Thank you, I forgot this was the definition of the `getData` function, not the function call.